src/d3d10/d3d10_device.cpp

```diff
@@ -8,9 +8,8 @@ namespace dxvk {
   D3D10Device::D3D10Device(
           D3D11Device*                      pDevice,
           D3D11ImmediateContext*            pContext)
-  : m_device(pDevice), m_context(pContext),
-    m_multithread(this, !(pDevice->GetCreationFlags() & D3D10_CREATE_DEVICE_SINGLETHREADED)) {
-    
+  : m_device(pDevice), m_context(pContext) {
+    m_context->GetMultithread()->SetMultithreadProtected(!(pDevice->GetCreationFlags() & D3D10_CREATE_DEVICE_SINGLETHREADED));
```

Comment:
This will enable multithreaded protection and thus cause a ~20% performance hit for **all** D3D11 apps as well, since the d3d10 device is always created. The correct fix would be to do this in `D3D10CoreCreateDevice`.